src/agent.c

```diff
@@ -467,8 +497,11 @@ int agent_relay_send(juice_agent_t *agent, agent_stun_entry_t *entry, const addr
 		JLOG_ERROR("STUN message write failed");
 		return -1;
 	}
-
-	return agent_direct_send(agent, &entry->record, buffer, size, ds);
+	if (agent_direct_send2(agent, entry, buffer, size, ds) < 0) {
+		JLOG_WARN("STUN message send failed, errno=%d", sockerrno);
```

Comment:
I cleaned up the `sockerrno` after `agent_direct_send()` as it's a bit clumsy (the function could internally make a last system call modifying errno), so you should do the same here.